filesystem/flysystem/FlysystemUrlPlugin.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace SilverStripe\Filesystem\Flysystem;
+
+use League\Flysystem\AwsS3v2\AwsS3Adapter;
+use League\Flysystem\Filesystem;
+use League\Flysystem\FilesystemInterface;
+use League\Flysystem\PluginInterface;
+use Oneup\FlysystemBundle\Adapter\LocalWithHost;
+
+
+/**
+ * Allows urls for files to be exposed
+ *
+ * Credit to https://github.com/SmartestEdu/FlysystemPublicUrlPlugin
+ */
+class FlysystemUrlPlugin implements PluginInterface {
+	
+    /**
+     * @var Filesystem adapter
+     */
+    protected $adapter;
+
+    public function setFilesystem(FilesystemInterface $filesystem) {
+        $this->adapter = $filesystem->getAdapter();
+    }
+
+    public function getMethod() {
+        return 'getPublicUrl';
+    }
+
+    /**
+	 * Generate public url
+	 * 
+     * @param string $path
+     * @return string The full url to the file
+     */
+    public function handle($path) {
+		// Default adaptor
+		if($this->adapter instanceof AssetAdaptor) {
+			return $this->adapter->getPublicUrl($path);
+		}
+
+		// Check S3 adaptor
+        if (class_exists('League\Flysystem\AwsS3v2\AwsS3Adapter')
```

Comment:
Is there a good reason for hard-coding a check for an S3 adapter here? Shouldn’t it be registered as a separate “plugin” or something? Seems weird to hard-code the `https://s3.amazonaws.com/%s/%s` like this